python_modules/dagster/dagster/_core/event_api.py

```diff
@@ -278,17 +278,22 @@ class AssetRecordsFilter(
 
     def __new__(
         cls,
-        asset_key: AssetKey,
+        asset_key: Optional[AssetKey] = None,
         asset_partitions: Optional[Sequence[str]] = None,
         after_timestamp: Optional[float] = None,
         before_timestamp: Optional[float] = None,
         after_storage_id: Optional[int] = None,
         before_storage_id: Optional[int] = None,
         storage_ids: Optional[Sequence[int]] = None,
     ):
+        check.invariant(
```

Comment:
raise some flavor of `DagsterError` instead of `check` so that this constraint is surfaced if its hit in cloud server context